app/javascript/packages/document-capture/components/in-person-location-step.tsx

```diff
@@ -30,9 +30,62 @@ interface FormattedLocation {
   weekdayHours: string;
 }
 
+interface PostOptions {
+  /**
+   * Whether to send the request as a JSON request. Defaults to inclusion.
+   */
+  json: boolean;
+
+  /**
+   * Whether to include CSRF token in the request. Defaults to inclusion.
+   */
+  csrf: boolean;
+
+  /**
+   * Optional. HTTP verb used. If excluded, will fetch as a GET.
+   */
+  method: string;
+}
+
+const getCsrfToken = () =>
+  document.querySelector<HTMLMetaElement>('meta[name="csrf-token"]')?.content;
+
+const request = async (
+  url: string,
+  body: BodyInit | object,
+  options: Partial<PostOptions> = {},
+) => {
+  const headers: HeadersInit = {};
+  const defaultOptions = {
+    csrf: true,
+    json: true,
+    ...options,
+  };
```

Comment:
The name "default options" seems a bit misleading, since we don't alter the options any more at any point after it's initialized here. I think it could make sense to extract `csrf` and `json` as a constant value of default options, but I'd expect it to look something more like...

```ts
const DEFAULT_OPTIONS = { csrf: true, json: true };

async function request(
  // ...
  options: Partial<PostOptions> = {},
) {
  // ...
  const mergedOptions: Partial<PostOptions> = { ...DEFAULT_OPTIONS, ...options };

  // ...
}
```